src/query/execute_statement.c

```diff
@@ -18056,3 +18056,103 @@ do_create_server (PARSER_CONTEXT * parser, PT_NODE * statement)
 
   return error;
 }
+
+int
+get_dblink_info_from_dbserver (PARSER_CONTEXT * parser, const char *server, DB_VALUE * out_val)
+{
+  char *server_name, *t;
+  DB_OBJECT *server_object, *server_class;
+  DB_IDENTIFIER server_obj_id;
+  DB_VALUE values[4];
+  int au_save, error, cnt;
+  bool is_au_disable = false;
+  const char *url_attr_names[4] = { "host", "port", "db_name", "properties" };
+
+  cnt = 0;
+  t = strchr ((char *) server, '.');	/* FIXME */
+  server_name = (t != NULL) ? (t + 1) : (char *) server;
+
+  server_class = sm_find_class (CT_DB_SERVER_NAME);
+  if (server_class == NULL)
+    {
+      error = ER_DBLINK_CATALOG_DB_SERVER_NOT_FOUND;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 0);
+      goto error_end;
+    }
+
+  server_object = do_get_server_obj_id (&server_obj_id, server_class, server_name);
+  if (server_object == NULL)
+    {
+      error = ER_DBLINK_SERVER_NOT_FOUND;
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, error, 1, server_name);
+      goto error_end;
+    }
+
+  AU_DISABLE (au_save);
+  is_au_disable = true;
```

Comment:
It seems redundant. AU_ENABLE should be called anyway at the end of this function

Reply:
I didn't understand which part seems to be redundant.
If there is no error, ENABLE is processed in advance at line 18126.
This is to reduce the time to leave the DISABLE state if possible.

Reply:
`is_au_disable` is redundant. just calling AU_ENABLE (au_save) is enough.

Reply:
Appropriate "au_save" values are required when performing AU_ENABLE().
au_save is obtained by performing AU_DISABLE().
However, in the code before AU_DISABLE (au_save) is executed, a "goto error_end" syntax may be encountered, but the value of au_save is not guaranteed in this case.
So we need a variable that indicates that the "au_save" value is guaranteed.

It is possible if the scope of leaving DISABLE state is expanded to the entire get_dblink_info_from_dbserver() function area.
I'll fix it like that (because it will run as a single process).